src/main/java/com/icthh/xm/tmf/ms/communication/messaging/AbstractDeliveryReportListener.java

```diff
@@ -35,6 +47,33 @@ public void onAcceptDeliverSm(DeliverSm deliverSm) {
         });
     }
 
+    protected DeliveryReceipt getShortMessage(DeliverSm deliverSm) {
+        try {
+            return deliverSm.getShortMessageAsDeliveryReceipt();
+        } catch (InvalidDeliveryReceiptException e) {
+            log.error("Cannot get short message, error: {}", e.getMessage());
+        }
+        return null;
+    }
+
+    protected String getMessageId(DeliveryReceipt deliveryReceipt){
+        String messageId = deliveryReceipt.getId();
+        try {
+            return ofNullable(messageId)
+                .map(id-> new BigInteger(id, 10))
```

Comment:
please extract magic numbers to named constants.

Reply:
extracted